### PR TITLE
feat(rust): add more commit info to most operations

### PR DESCRIPTION
**Pull request by ion-elgreco**
# Description
Some operations were not writing the readVersion and operationMetrics to the commitInfo. This adds that to those operations as well.

### CODE REVIEW

**Comment by ion-elgreco** [comment]
@roeap I can also undo the change of the commit in restore, then we can look at that another time?

**Comment by ion-elgreco** [comment]
@rtyler @roeap I've removed the high level commit and reverted back to the old version for restore : )

File: crates/deltalake-core/src/operations/delete.rs [hunk 1]
```diff
@@ -249,6 +249,19 @@ async fn execute(
...
+    };
+
+    app_metadata.insert("readVersion".to_owned(), snapshot.version().into());
+
+    let delete_metrics = serde_json::to_value(metrics.clone());
+
+    if let Ok(map) = delete_metrics {
+        app_metadata.insert("operationMetrics".to_owned(), map);
+    }
+
```
Comment (roeap) [inline crates/deltalake-core/src/operations/delete.rs:264]: I think we can avoid cloning here.

```suggestion
    app_metadata.insert("readVersion".to_owned(), snapshot.version().into());

    if let Ok(map) = serde_json::to_value(&metrics) {
        app_metadata.insert("operationMetrics".to_owned(), map);
    }

```

**Review by roeap — COMMENTED** [review]
Looking good so far.

I do think we can avoid cloning these metrics, and with that also don't need the `Clone` implementations.

**Review by roeap — APPROVED** [review]

**Pull request merged**